cmd/validate_models.go

```diff
@@ -0,0 +1,161 @@
+package cmd
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/openfga/openfga/cmd/util"
+	"github.com/openfga/openfga/pkg/storage"
+	"github.com/openfga/openfga/pkg/storage/mysql"
+	"github.com/openfga/openfga/pkg/storage/postgres"
+	"github.com/openfga/openfga/pkg/storage/sqlcommon"
+	"github.com/openfga/openfga/pkg/typesystem"
+	"github.com/spf13/cobra"
+	"github.com/spf13/viper"
+)
+
+func NewValidateCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "validate-models",
+		Short: "Validate authorization models. NOTE: this command is in beta and may be removed in future releases.",
+		Long:  "List all authorization models across all stores and run validations against them.\nNOTE: this command is in beta and may be removed in future releases.",
+		RunE:  runValidate,
+		Args:  cobra.NoArgs,
+		PreRun: func(cmd *cobra.Command, args []string) {
+			flags := cmd.Flags()
+
+			util.MustBindPFlag(datastoreEngineFlag, flags.Lookup(datastoreEngineFlag))
+			util.MustBindPFlag(datastoreURIFlag, flags.Lookup(datastoreURIFlag))
+		},
+	}
+
+	flags := cmd.Flags()
+	flags.String(datastoreEngineFlag, "", "the datastore engine")
+	flags.String(datastoreURIFlag, "", "the connection uri to the datastore")
+
+	// NOTE: if you add a new flag here, add the binding in PreRunE
+
+	return cmd
+}
+
+type ValidationResult struct {
```

Comment:
Was making this publicly exported done on purpose? What's the use case for making a public contract on this? Can't we make this an unexported type that is only used in the scope of this CLI entrypoint?

Reply:
Not on purpose 😅